app/models/user.rb

```diff
@@ -3,5 +3,26 @@ class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable
+ def self.connect_to_linkedin(auth, signed_in_resource=nil)
+   user = User.where(:provider => auth.provider, :uid => auth.uid).first
+   if user
+     return user
+   else
+     registered_user = User.where(:email => auth.info.email).first
+     if registered_user
+       return registered_user
+     else
+
+       user = User.create(name:auth.info.first_name,
+                           provider:auth.provider,
+                           uid:auth.uid,
+                           email:auth.info.email,
+                           password:Devise.friendly_token[0,20],
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Space missing after colon.<br>Space missing after comma.